---
title: "Genomic technologies"
author: "Mikhail Dozmorov mdozmorov@vcu.edu"
date: "May 15, 2019"
output:
  beamer_presentation:
    colortheme: seahorse
    # colortheme: dolphin
    fig_caption: no
    fig_height: 6
    fig_width: 7
    fonttheme: structurebold
    # theme: boxes
    theme: AnnArbor
---

## Sequencing in a nutshell

- Cut the long DNA into smaller segments (several hundreds to several thousand bases).
- Sequence each segment: start from one end and sequence along the chain, base by base.
- The process stops after a while because the noise level is too high.
- Results from sequencing are many sequence pieces. The lengths vary, usually a few thousands from Sanger, and several hundreds from NGS.
- The sequence pieces are called "reads" for NGS data.

## Two shotgun-sequencing strategies

\begin{center}
\includegraphics[height=200px]{img/shotgun_strategies.png}
\end{center}

\tiny https://www.nature.com/nrg/journal/v2/n8/full/nrg0801_573a.html

<!--a | Schematic overview of clone-by-clone shotgun sequencing. A representation of a genome is made by analogy to an encyclopaedia set, with each volume corresponding to an individual chromosome. The construction of clone-based physical maps produces overlapping series of clones (that is, contigs), each of which spans a large, contiguous region of the source genome. Each clone (for example, a bacterial artificial chromosome (BAC)) can be thought of as containing the DNA represented by one page of a volume. For shotgun sequencing, individual mapped clones are subcloned into smaller-insert libraries, from which sequence reads are randomly derived. In the case of BACs, this typically requires the generation of several thousand sequence reads per clone. The resulting sequence data set is then used to assemble the complete sequence of that clone (see Figs 3,4). b | Schematic overview of whole-genome shotgun sequencing. In this case, the mapping phase is skipped and shotgun sequencing proceeds using subclone libraries prepared from the entire genome. Typically, tens of millions of sequence reads are generated and these in turn are subjected to computer-based assembly to generate contiguous sequences of various sizes.-->

## The competing human genome projects

\begin{center}
\includegraphics[height=180px]{img/sequencing_race.png}
\end{center}

## Evolution of sequencing technologies

- 2006: Solexa (Illumina)
- 2010: Ion Torrent (Life Technologies)
- 2011: Pacific Biosciences
- 2015: Oxford Nanopore Technologies

## Solexa (Illumina) sequencing (2006)

- Polymerase Chain Reaction (PCR) amplify DNA fragments
- Immobilize fragments on a solid surface, amplify
- Reversible terminator sequencing with 4 color dye-labelled nucleotides

\tiny Video of Illumina sequencing, http://www.youtube.com/watch?v=77r5p8IBwJk (1.5m), https://www.youtube.com/watch?v=fCd6B5HRaZ8 (5m)

## Solexa (Illumina) sequencing (2006)

\begin{center}
\includegraphics[height=220px]{img/Cluster_Generation.png}
\end{center}

\tiny http://www.annualreviews.org/doi/abs/10.1146/annurev.genom.9.081307.164359

##  Cluster amplification by "bridge" PCR

\begin{center}
\includegraphics[height=150px]{img/illumina_bridge_pcr.png}
\end{center}

\tiny https://binf.snipcademy.com/lessons/ngs-techniques/bridge-pcr

## Clonal amplification

\begin{center}
\includegraphics[height=200px]{img/illumina_cluster_amplification.png}
\end{center}

## Base calling

- 6 cycles with base-calling

\begin{center}
\includegraphics[height=150px]{img/illumina_base_calling.png}
\end{center}

\tiny https://www.youtube.com/watch?v=IzXQVwWYFv4

https://www.youtube.com/watch?time_continue=65&v=tuD-ST5B3QA

## Illumina sequencers

\begin{center}
\includegraphics[height=100px]{img/HiSeq_X_Five_Sequencing_System.jpg}
\end{center}

- **Illumina HiSeq**: ~3 billion paired 100bp reads, ~600Gb, $10K, 8 days (or "rapid run" ~90Gb in 1-2 days)
- **Illumina X Ten**: ~6 billion paired 150bp reads, 1.8Tb, <3 days, ~1000 / genome(\$\$), (or "rapid run" ~90Gb in 1-2 days)
- **Illumina NextSeq**: One human genome in <30 hours

\tiny http://www.businesswire.com/news/home/20150112006333/en/Illumina-Expands-World%E2%80%99s-Comprehensive-Next-Generation-Sequencing-Portfolio


## Illumina sequencers

\begin{center}
\includegraphics[height=130px]{img/Illumina_NovaSeq6000.png}
\end{center}

- Massive improvement of the cluster density - higher output
- Less expensive than the previous sequencers
- Faster runs

\tiny https://blog.genohub.com/2017/01/10/illumina-unveils-novaseq-5000-and-6000/

http://www.mrdnalab.com/illumina-novaseq.html

<!-- http://www.opiniomics.org/hiseq-move-over-here-comes-nova-a-first-look-at-illumina-novaseq/ -->

## Solexa (Illumina) sequencing: summary

Advantages:

- Best throughput, accuracy and read length
- Fast & robust library preparation

Disadvantages:

- Inherent limits to read length (practically, 150bp)
- Some runs are error prone

\tiny Video of Illumina sequencing https://www.youtube.com/watch?v=womKfikWlxM (5m)

## Single-end vs. paired-end sequencing

- Single-end sequencing: sequence one end of the DNA segment.
- Paired-end sequencing: sequence both ends of a DNA segments.
    - Result reads are "paired", separated by certain length (the length of the DNA segments, usually a few hundred bps).
    - Paired-end data can be used as single-end, but contain extra information which is useful in some cases, e.g., detecting structural variations in the genome.
    - Modeling technique is more complicated.

## Paired-end sequencing - a workaround to sequence longer fragments

- Read one end of the molecule, flip, and read the other end
- Generate pair of reads separated by up to 500bp with inward orientation

\begin{center}
\includegraphics[height=150px]{img/illumina_paired_end.png}
\end{center}

## ION Torrent-pH Sensing of Base Incorporation

\begin{center}
\includegraphics[height=170px]{img/iontorrent.png}
\end{center}

## Platforms: Ion Torrent

\begin{center}
\includegraphics[height=140px]{img/ion_platforms.png}
\end{center}

- Low substitution error rate, in/dels problematic, no paired end reads
- Inexpensive and fast turn-around for data production
- Improved computational workflows for analysis

##  Pacific Biosciences

\begin{center}
\includegraphics[height=150px]{img/pacbio.jpg}
\end{center}

- Long reads
    - Structural variant discovery
    - _De novo_ genome assembly

\tiny https://www.forbes.com/forbes/2009/1005/revolutionaries-science-genomics-gene-machine.html

## Pacific Biosciences: summary

Key Points:

- 1 DNA molecule and 1 polymerase in each well (zero-mode waveguide)
- 4 colors flash in real time as polymerase acts
- Methylated cytosine has distinct pattern
- No _theoretical_ limit to DNA fragment length

Caveats:

- Higher error rate (11-15%), but they are random
- Lower throughput, roughly 5 gigabases per run

## Nanopore sequencing

- Nearly 30-years old technology

\begin{center}
\includegraphics[height=160px]{img/nanopore_x616[1].jpg}
\end{center}

\tiny http://www2.technologyreview.com/news/427677/nanopore-sequencing/

## Nanopore sequencing

- Nanopore sequencing with ONT is accurate and relatively reliable
- Current yield per run ("R9.4" chemistry): ~5 Gbp, 5-15% error rate

\begin{center}
\includegraphics[height=130px]{img/nanoporex2760.jpg}
\end{center}

\tiny https://www.technologyreview.com/s/600887/with-patent-suit-illumina-looks-to-tame-emerging-british-rival-oxford-nanopore/

Video of Ion Torrent chemistry, http://www.youtube.com/watch?v=yVf2295JqUg (2.5m)

## Developments in next generation sequencing: instruments, read lengths, throughput.

\begin{center}
\includegraphics[height=190px]{img/developments_in_high_throughput_sequencing.jpg}
\end{center}

\tiny https://github.com/lexnederbragt/developments-in-next-generation-sequencing


# Sequencing applications

## Applications

- NGS has a wide range of applications.
    - DNA-seq: sequence genomic DNA.
    - RNA-seq: sequence RNA products.
    - ChIP-seq: detect protein-DNA interaction sites.
    - Bisulfite sequencing (BS-seq): measure DNA methylation strengths.
    - A lot of others.
- Basically replaced microarrays with better data: greater dynamic range and higher signal-to-noise ratios.

## What matters is what you feed into the sequencing machine

\includegraphics[height=190px]{img/seq_pachter.png}

\tiny https://liorpachter.wordpress.com/seq/

## Evolution of sequencing technologies

\includegraphics[height=200px]{img/seq_technologies.png}

# RNA-seq

## What is RNA sequencing?

- Massive parallel sequencing to **characterize and quantify transcriptomes** (all actively transcribed genes)
- Detection of **differential gene expression**
- **Transcriptome reconstruction**, identification of **new transcripts**
- Detection of **alternative splicing events**
- Detection of **structural variants**, e.g., fusion transcripts
- **Allele-specific** gene expression measurements
- **Mutation analysis** – presence of genomic mutations and their effect on gene expression

\tiny http://journals.plos.org/ploscompbiol/article/file?type=supplementary&id=info:doi/10.1371/journal.pcbi.1004393.s003

## Overview of RNA sequencing technology

\includegraphics[height=200px]{img/rna-seq_tech.png}

\tiny Source: http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1004393

## Library preparation steps

- **RNA isolation and QC**, to extract RNA relevant to the experimental question
- **Fragmentation**, to recover short reads across full length of long genes
- **Size selection**, suitable for RNA sequencing. 300-500bp - mRNA, 20-150bp - small/miRNA
- **Amplification**, typically by PCR. Up to $0.5-10ng$ of RNA
- **Library normalization/Exome capture**
- **Barcoding** and **multiplexing**
- Optionally, add **External RNA Control Consortium (ERCC) spike-in controls**
- **Single** or **paired end** sequencing. The latter is preferrable for the _de novo_ transcript discovery or isoform expression analysis

\tiny Sample prepatation and library construction strategies: http://journals.plos.org/ploscompbiol/article/file?type=supplementary&id=info:doi/10.1371/journal.pcbi.1004393.s005

## RNA isolation

- **Ribosomal RNA (rRNA) depletion**
    - $0.1-1\mu g$ original total RNA (One cell contains ~10 picogram of total RNA)
    - rRNAs constitute over 90 % of total RNA in the cell, leaving the 1–2 % comprising messenger RNA (mRNA) that we are normally interested in (One cell contains ~0.1 picogram mRNA)
    - Enriches for mRNA + long noncoding RNA. 
    - Hybridization to bead-bound rRNA probes

## RNA isolation

- **Poly(A) selection (for eukaryotes only)**
    - Enrich for mRNA.
    - Hybridization to oligo-dT beads

- **Small RNA extraction**
    - Specific kits required to retain small RNAs
    - Optionally, size-selection by gel

&nbsp;

\tiny Description of RNA-seq library enrichment strategies: http://journals.plos.org/ploscompbiol/article/file?type=supplementary&id=info:doi/10.1371/journal.pcbi.1004393.s006

## RNA quality

Agilent 2100 bioanalyzer. RIN - RNA integrity number (should be >7)

\includegraphics[height=200px]{img/agilent.png}

## Unstranded vs. Strand-specific library

- **Unstranded**:  Random hexamer priming to reverse-transcribe mRNA
- **Stranded**:  dUTP method - incorporating UTP nucleotides during the second cDNA synthesis, followed by digestion of the strand containing dUTP

&nbsp;

\tiny Strand-related settings for RNA-seq tools: http://journals.plos.org/ploscompbiol/article/file?type=supplementary&id=info:doi/10.1371/journal.pcbi.1004393.s007}

## Sequencing length/depth

- Longer reads improve mappability and transcript quantification
- More transcripts will be detected and their quantification will be more precise as the sample is sequenced to a deeper level
- Up to 100 million reads is needed to precisely quantify low expressed transcripts. 
- In reality, 20-30 million reads is OK for human genome.

## Summarization of read counts

- From RNA-seq, the alignment result gives the chromosome/position of each aligned read.
- For a gene, there are reads aligned to the gene body. How to summarized them into a number for the expression?

**Counts of reads**

- Easiest: The relative expression of a transcript is proportional to the number of cDNA fragments that originate from it ~ number of aligned reads. 
- Disadvantages: longer gene produce more reads, library depth (total counts) influence counts of individual transcripts
- **htseq-count** - command line tool for counting reads in features
- **featureCounts** - summarize multiple datasets at the same time

\tiny https://htseq.readthedocs.io/en/release_0.9.1/count.html

http://bioinf.wehi.edu.au/featureCounts/

## Expression estimation for known genes and transcripts

- **Counts per million**: counts scaled by the library depth in million units. $CPM=C * 10^6 / N$
- **RPKM**: Reads Per Kilobase of transcript per Million mapped reads. Introduced by Mortazavi, 2008
- **FPKM**: Fragments Per Kilobase of transcript per Million mapped reads. Introduced by Salzberg, Pachter, 2010

## Expression estimation for known genes and transcripts

- **FPKM** (or **RPKM**) attempt to normalize for gene size and library depth

$$RPKM\ (or\ FPKM_i)=(10^9*C_i)/(N*L_i)$$

- $C_i$ - number of mappable reads/fragments for a $i$ gene/transcript/exon/etc.
- $N$ - total number of mappable reads/fragments in the library
- $L_i$ - number of base pairs in the $i$ gene/transcript/exon/etc.

\tiny https://haroldpimentel.wordpress.com/2014/05/08/what-the-fpkm-a-review-rna-seq-expression-units/

## TPM: Transcript per Kilobase Million

- **TPM**: Transcripts per million. Introduced by Li, 2011. Normalized by total transcript count instead of read count in addition to average read length.

If you were to sequence one million full length transcripts, TPM is the number of transcripts you would have seen for transcript $i$.

$$TPM_i = 10^6 * Z * \frac{C_i}{N*L_i}$$

- $Z$ - sum of all length normalized transcript counts

## TPM: Transcript per Kilobase Million

**FPKM is calculated as**

1. Sum sample/library fragments per million
2. Divide gene/transcript fragment counts by #1 –  fragments per million, FPM
3. Divide FPM by length of gene in kilobases (FPKM)

**TPM reverses the order - length first, library size second**

1. Divide fragment count by length of transcript –  fragments per kilobase, FPK
2. Sum all FPK for sample/library per million
3. Divide #1 by #2 (TPM)

\tiny https://youtu.be/TTUrtCY2k-w?t=23  

https://www.ncbi.nlm.nih.gov/pubmed/22872506


# Single cell sequencing

## Single cell sequencing applications

- Infer cell lineages
- Identify subpopulations
- Outline temporal evolution
- Define cell-specific biological characteristics, e.g., differentially expressed genes

## Single-cell Sequencing Technology

\begin{center}
\includegraphics[height=140px]{img/technology.png}
\end{center}

A single device has three input ports (oil, barcoded beads in lysis buffer, and cells of interest) and a single output port used for collecting bead–cell-containing lipid droplets. Then each cell (or RNA in the cell) is marked by the unique barcode and processed on the bead for sequencing.

## Timeline of single-cell technologies

\begin{center}
\includegraphics[height=100px]{img/scrna_seq_timeline.png}
\end{center}

\tiny Svensson, Valentine, Roser Vento-Tormo, and Sarah A. Teichmann. “Exponential Scaling of Single-Cell RNA-Seq in the Past Decade.” Nature Protocols 13, no. 4 (April 2018): 599–604. https://doi.org/10.1038/nprot.2017.149.

## The computational workflow for single cell experiments detailed in steps

\begin{center}
\includegraphics[height=190px]{img/workflow_computational.png}
\end{center}

## How does single-cell data differ from bulk RNA-seq

- In theory, up to 600 cells could be sequenced on a single lane (assuming 120 million reads per lane)
- Even with the most sensitive platforms, the data are relatively sparse owing to a high frequency of dropout events (lack of detection of specific transcripts)
- The numbers of expressed genes detected from single cells are typically lower compared with population-level ensemble measurements
- The commonly used ‘reads per kilobase per million’ (RPKM) transcript quantification is biased on a single-cell level, at the very least the ‘transcripts per million’ (TPM) should be used

## Abundance of zeros

\begin{center}
\includegraphics[height=150px]{img/prop_zeros.png}
\end{center}

\tiny Bacher, Rhonda, and Christina Kendziorski. “Design and Computational Analysis of Single-Cell RNA-Sequencing Experiments.” Genome Biology 17 (April 7, 2016): 63. https://doi.org/10.1186/s13059-016-0927-y.

<!--Prominent features in single-cell RNA-seq data relative to bulk RNA-seq include an abundance of zeros, increased variability, and multi-modal expression distributions. a Boxplots of the gene-specific proportion of zeros in a bulk (bulk1) and single-cell (sc1) dataset stratified by percentile of median gene expression. Sequencing depth ranges from 420,000 to 16.6 million in bulk1 and 385,000 to 16.4 million in sc1 (samples were chosen to have comparable depths; see the “Data” section).-->

## How does single-cell data differ from bulk RNA-seq

- scRNA-seq data, in general, are much more variable than bulk data
- Distributions of transcript quantities are often more complex in single-cell datasets than in bulk RNA-seq - negative binomial or multimodal distributions

## Filtering

- Filter cells and/or genes
- No single consensus, frequently used criteria include:
    - relative library size
    - number of detected genes (< 10,000)
    - fraction of reads mapping to mitochondria-encoded genes or synthetic spike-in RNAs (< 500)

## Global-scaling normalization

\begin{center}
\includegraphics[height=110px]{img/global_normalization.png}
\end{center}

\tiny Vallejos, Catalina A, Davide Risso, Antonio Scialdone, Sandrine Dudoit, and John C Marioni. “Normalizing Single-Cell RNA Sequencing Data: Challenges and Opportunities.” Nature Methods 14, no. 6 (May 15, 2017): 565–71. https://doi.org/10.1038/nmeth.4292.

## Sub-population identification

Standard methods used in RNA-Seq

- **Hierarchical clustering**, **PCA**, **tSNE** of highly variable, or differentially expressed, genes. Zeros can be a problem
- **ZIFA** - Zero-inflated dimensionality reduction algorithm for single-cell data
- **SNN-Cliq** - A clustering method for high dimensional dataset. Rank-based (not expression) similarity

\tiny https://github.com/epierson9/ZIFA

http://bioinfo.uncc.edu/SNNCliq/

Many more at https://genomebiology.biomedcentral.com/articles/10.1186/s13059-016-0927-y

## Cell clustering

- Perhaps the most active topic in scRNA-seq
- The goals include:
    - Cluster cells into subgroups
    - Model temporal transcriptomic dynamics: reconstruct “pseudo-time” for cells. This is useful for understanding development or disease progression
- Traditional method like k-means or hierarchical clustering need to be used with caution due to dropout events

## t-SNE: a useful visualization tool

- t-SNE (t-distributed stochastic neighbor embedding): visualize high-dimensional data on 2-/3-D map
- When project high-dimensional data into lower dimensional space, preserve the distances among data points
    - This alleviate the problem that many clusters overlap on low dimensional space
- Try to make the pairwise distances of points similar in high and low dimension
- This is used in almost all scRNA-seq data visualization
- Has “tsne” package on CRAN

## Pseudotemporal ordering 

- Idea - cells at different differentiation (or other biological process) stage are presented with different expression profiles
- Dynamics of cellular processes can be reconstructed from expression profiles
- Key assumption: genes do not change direction very often, thus samples with similar transcriptional profiles should be close in order
- Most approaches are dimensionality reduction-based, and apply graph theory designed to traverse nodes in a graph efficiently
- **Monocle** - Independent component analysis, then a minimum spanning three through the dimension-reduced data

\tiny https://cole-trapnell-lab.github.io/monocle-release/

Many more at https://github.com/agitter/single-cell-pseudotime

## Monocle, An analysis toolkit for single-cell RNA-seq

Single-cell trajectories, clustering, visualization, differential expression

\begin{center}
\includegraphics[height=170px]{img/monocle.png}
\end{center}

\tiny https://cole-trapnell-lab.github.io/monocle-release/

Trapnell, Cole, Davide Cacchiarelli, Jonna Grimsby, Prapti Pokharel, Shuqiang Li, Michael Morse, Niall J. Lennon, Kenneth J. Livak, Tarjei S. Mikkelsen, and John L. Rinn. “The Dynamics and Regulators of Cell Fate Decisions Are Revealed by Pseudotemporal Ordering of Single Cells.” Nature Biotechnology 32, no. 4 (April 2014): 381–86. https://doi.org/10.1038/nbt.2859.

## Differentially expressed genes

- Need to accomodate unobserved dropouts, bimodality in expression levels due to abundance of zero or low values (**MAST**, **SCDE**)
- **scDD** - Distinguishes four types of differential expression changes to increase power:
    - shifts in unimodal distribution
    - differences in the number of modes
    - differences in the proportion of cells within modes
    - combination of the previous two

\tiny https://github.com/kdkorthauer/scDD

# Functional enrichment analysis

## Why enrichment analysis? 

- Translating changes of **hundreds/thousands of differentially expressed genes** into a few biological processes (reducing dimensionality)
- High level understanding of the biology behind gene expression – **Interpretation!**

\begin{center}
\includegraphics[height=90px]{img/why_enrichment2.png}
\end{center}

## What is enrichment analysis

- **Enrichment analysis** - summarizing common functions associated with a group of objects

\begin{center}
\includegraphics[height=120px]{img/enrichment_analysis.png}
\end{center}

## Annotation databases and ontologies

- An annotation database annotates genes with functions or properties - sets of genes with shared functions
- Structured prior knowledge about genes

\begin{center}
\includegraphics[height=140px]{img/GO_db.png}
\end{center}

## Gene ontology structure

Gene ontology describes multiple levels of detail of gene function.

- **Molecular Function** - the tasks performed by individual gene products; examples are _transcription factor_ and _DNA helicase_ 
- **Biological Process** - broad biological goals, such as _mitosis_ or _purine metabolism_, that are accomplished by ordered assemblies of molecular functions 
- **Cellular Component** - subcellular structures, locations, and macromolecular complexes; examples include _nucleus_, _telomere_, and _origin recognition complex_

## Other gene annotation databases

- **KEGG: Kyoto Encyclopedia of Genes and Genomes** - a collection of biological information compiled from published material, http://www.genome.jp/kegg/
- **Reactome** - Curated human pathways encompassing metabolism, signaling, and other biological processes, http://www.reactome.org/
- **MSigDb** - Molecular Signatures Database, http://software.broadinstitute.org/gsea/msigdb/
\small
- **PathwayCommons**, version 8 has over 42,000 pathways from 22 data sources, http://www.pathwaycommons.org/
- **PathGuide**, lists ~550 pathway related databases, http://www.pathguide.org/
- **WikiPathways**, community-curated pathways, http://wikipathways.org/
- **BioCarta**, pathway genes and diagrams, https://cgap.nci.nih.gov/Pathways/BioCarta_Pathways
- **Consensus-PathDB**, pathway interactions, enrichment, data, http://www.consensuspathdb.org/

## Overrepresentation analysis, Hypergeometric test

- $m$ is the total number of genes
- $j$ is the number of genes are in the functional category
- $n$ is the number of differentially expressed genes
- $k$ is the number of differentially expressed genes in the category

\begin{center}
\includegraphics[height=120px]{img/hypergeometric.png}
\end{center}

## Overrepresentation analysis, Hypergeometric test

- $m$ is the total number of genes
- $j$ is the number of genes are in the functional category
- $n$ is the number of differentially expressed genes
- $k$ is the number of differentially expressed genes in the category

What is the probability of having $k$ or more genes from the category in the selected $n$ genes?

$$P = \sum_{i=k}^n{\frac{\binom{m-j}{n-i}\binom{j}{i}}{{m \choose n}}}$$

## Overrepresentation analysis, Hypergeometric test

1. Find a set of differentially expressed genes (DEGs)
2. Are _DEGs in a set_ more common than _DEGs not in a set_?

- Fisher test `stats::fisher.test()`
- Conditional hypergeometric test, to account for directed hierachy of GO `GOstats::hyperGTest()`

\tiny Example: https://github.com/mdozmorov/MDmisc/blob/master/R/gene_enrichment.R

## GSEA: Gene set enrichment analysis

- **Gene set analysis (GSA)**. Mootha et al., 2003; modified by Subramanian, et al. "**Gene set enrichment analysis: A knowledge-based approach for interpreting genome-wide expression profiles.**" PNAS 2005 http://www.pnas.org/content/102/43/15545.abstract
- Main rationale – functionally related genes often display a coordinated expression to accomplish their roles in the cells
- Aims to identify gene sets with "subtle but coordinated"  expression changes that would be missed by DEGs threshold selection

## GSEA: Gene set enrichment analysis

1. Sort genes by log fold change
2. Calculate running sum - increment when gene in a set, decrement when not
3. Maximum of the runnig sum is the enrichment score - larger means genes in a set are toward top of the sorted list
4. Permute subject labels to calculate significance p-value

\begin{center}
\includegraphics[height=120px]{img/gsea.png}
\end{center}

## Gene set enrichment analysis

- **GSEA** (https://www.broadinstitute.org/gsea/index.jsp) - Better way of doing enrichment analysis
- **g:Profiler** (http://biit.cs.ut.ee/gprofiler/) - gene ID converter, GO and pathway enrichment, and more
- **ToppGene** (https://toppgene.cchmc.org) - Quick gene enrichment analysis in multiple categories
- **Metascape** (http://metascape.org/) - Enrichment analysis of multiple gene sets
- **DAVID** (https://david.ncifcrf.gov/) - Newly updated gene enrichment analysis
- **FRY** (http://shiny.bioinf.wehi.edu.au/giner.g/FRY_GeneSetExplorerApp/) - Fast Interactive Biological Pathway Miner, from WEHI group 

## R packages: enrichment analysis

- **clusterProfiler** (https://bioconductor.org/packages/release/bioc/html/clusterProfiler.html) - statistical analysis and visualization of functional profiles for genes and gene clusters
- **limma** (https://bioconductor.org/packages/release/bioc/html/limma.html) - Linear Models for Microarray Data, includes functional enrichment functions `goana`, `camera`, `roast`, `romer`
- **GOstats** (https://www.bioconductor.org/packages/2.8/bioc/html/GOstats.html) - tools for manimpuating GO and pathway enrichment analyses. https://github.com/mdozmorov/MDmisc/blob/master/R/gene_enrichment.R

## R packages: Gene annotation databases

- **annotables** (https://github.com/stephenturner/annotables) - R data package for annotating/converting Gene IDs
- **msigdf** (https://github.com/stephenturner/msigdf) - Molecular Signatures Database (MSigDB) in a data frame
- **pathview** (https://www.bioconductor.org/packages/devel/bioc/html/pathview.html) - a tool set for pathway based data integration and visualization


# ChIP (Chromatin Immunoprecipitation) sequencing

## ChIP-seq

- Chromatin-Immunoprecipitation (ChIP) followed by sequencing (seq): sequencing version of ChIP-chip.
- Used to detect locations of certain "events" on the genome:
    - Transcription factor binding.
    - DNA methylations and histone modifications.
- A type of "captured" sequencing. ChIP step is to capture genomic regions of interest.

## Gene Regulation: DNA -> RNA -> Protein 

- What are the transcription factors (TFs) that control gene expression? 
- At what genes do these TFs operate?
- Understanding transcriptional regulatory network will
    - Reveal how cellular processes are connected and coordinated
    - Suggest new strategies to manipulate phenotypes and combat disease

\begin{center}
\includegraphics[height=100px]{img/tf_regulation.png}
\end{center}

##  ChIP-seq big picture

Combine high-throughput sequencing with Chromatin Immunoprecipitation to identify specific protein-DNA interactions genome-wide, including those of:

- Transcription factors
- Histones (various types and modifications)
- RNA Polymerase (survey of transcription)
- DNA polymerase (investigate DNA replication) 
- DNA repair enzymes
- Fragments of DNA that are modified (e.g. methylated)

## Experimental procedures

1. **Crosslink**: fix proteins on Isolate genomic DNA.
2. **Sonicate**: cut DNA in small pieces of ~200bp.
3. **Immunoprecipitate (IP)**: use antibody to capture DNA segments with specific proteins.
4. **Reverse crosslink**: remove protein from DNA.
5. **Sequence** the DNA segments.

## ChIP-seq overview

\begin{center}
\includegraphics[height=200px]{img/chip-seq_overview.png}
\end{center}

## ChIP-seq "peak" detection

- When plot the read counts against genome coordinates, the binding sites show a tall and pointy peak. So "peaks" are used to refer to protein binding or histone modification sites

\begin{center}
\includegraphics[height=50px]{img/chip-seq-peaks.jpg}
\end{center}

- Peak detection is the most fundamental problem in ChIP-seq data analysis

##  Peak calling: a classic signal versus noise problem

\begin{center}
\includegraphics[height=190px]{img/chip-seq_input.png}
\end{center}

\tiny Park, Peter J. “ChIP–seq: Advantages and Challenges of a Maturing Technology.” Nature Reviews Genetics 10, no. 10 (October 2009): 669–80. https://doi.org/10.1038/nrg2641.

## Control sample is important

- A control sample is necessary for correcting many artifacts: DNA sequence dependent artifacts, chromatin structure, repetitive regions, etc.
- Importantly, control samples should be sequenced significantly deeper than the ChIP ones in a TF experiment and in experiments involving diffused broad-domain chromatin data. This is to ensure sufficient coverage of a substantial portion of the genome and non-repetitive autosomal DNA regions

\begin{center}
\includegraphics[height=80px]{img/chip-seq-peaks_control.jpg}
\end{center}

## Control sample is important

- There are three commonly used types of control sample: 
    1) **Input DNA** (a portion of the DNA sample removed prior to immunoprecipitation (IP))
    2) **Mock IP DNA** (DNA obtained from IP without antibodies)
    3) **DNA from nonspecific IP** (IP performed using an antibody, such as immunoglobulin G, against a protein that is not known to be involved in DNA binding or chromatin modification)

## How do peak-finders map binding sites?

- Fragments contain the TF binding site at a (mostly) random position within them
- Reads are randomly generated from left or right edges (sense or antisense) of fragments
- Binding site position = mid-way between sense tag peak and antisense tag peak
- To get binding site peak, shift sense downstream by 1⁄2 fragsize & antisense upstream by 1⁄2 fragsize

\begin{center}
\includegraphics[height=70px]{img/chip_peak_detection.png}
\end{center}

\tiny Kharchenko, Peter V, Michael Y Tolstorukov, and Peter J Park. “Design and Analysis of ChIP-Seq Experiments for DNA-Binding Proteins.” Nature Biotechnology 26, no. 12 (December 2008): 1351–59. https://doi.org/10.1038/nbt.1508.

## ChIP-seq for histone modification

- Histone modifications have various patterns
- Some are similar to protein binding data, e.g., with tall, sharp peaks: H3K4
- Some have wide (mega-bp) "blocks": H3K9
- Some are variable, with both peaks and blocks: H3K27me3, H3K36me3. Also, RNA Pol II - stalled binding shows as peaks, moving along with transcription shows as broad stretches

## Histone modification ChIP-seq data

\begin{center}
\includegraphics[height=180px]{img/histone_peaks.png}
\end{center}

## After peak/block calling

\begin{center}
\includegraphics[height=200px]{img/chip_downstream.png}
\end{center}

## Other "captured/targeted" sequencing technologies

- Enrich and then sequence selected genomic regions.
    - **MeDIP-seq**: measure methylated DNA.
    - **DNase-seq**: detect DNase I hypersensitive sites.
    - **FAIRE-seq**: detect open chromatin sites.
    - **Hi-C**: study 3D structure of chromatin conformation.
    - **GRO-seq**: map the position, amount and orientation of transcriptionally engaged RNA polymerases.
    - **Ribo-seq**: detect ribosome occupancy on mRNA. This is captured RNA- seq.

## DNAse-seq

- A widely used approach in gene regulation studies uses DNase I as a tool to identify DNase I Hypersensitive Sites (DHSs) within chromatin
- DHSs represent open chromatin regions that are normally only accessible at sites of active regulatory elements such as transcriptional enhancers

\begin{center}
\includegraphics[height=120px]{img/dnase.png}
\end{center}

\tiny Cockerill,P.N. (2011) Structure and function of active chromatin and DNase I hypersensitive sites. FEBS J., 278, 2182–2210.

<!--Figure from https://academic.oup.com/nar/article/41/21/e201/1283596-->

## ATAC-seq: finding open chromatin regions

\begin{center}
\includegraphics[height=160px]{img/atacseq_vs_others.png}
\end{center}

\tiny Jason D Buenrostro et al., “Transposition of Native Chromatin for Fast and Sensitive Epigenomic Profiling of Open Chromatin, DNA-Binding Proteins and Nucleosome Position,” Nature Methods 10, no. 12 (December 2013): 1213–18, https://doi.org/10.1038/nmeth.2688.

## Technology-specific data

Peaks produced by different technologies are different - calling peaks should be adjusted

\begin{center}
\includegraphics[height=170px]{img/mnase_dnase_atac.jpg}
\end{center}

\tiny https://www.the-scientist.com/?articles.view/articleNo/44772/title/Reveling-in-the-Revealed/

https://www.biostars.org/p/209592/

## Learn more (https://github.com/mdozmorov/Talk_Genomics)

\begin{center}
\includegraphics[height=150px]{img/bioinformatics_resources.jpg}
\end{center}

\tiny Statistical Methods for High-throughput Genomic Data I, II: https://mdozmorov.github.io/BIOS567.2017/syllabus/, https://mdozmorov.github.io/BIOS668.2018/syllabus/

Collections of bioinformatics resources: https://github.com/mdozmorov/blogs/tree/master/Bioinformatics

https://www.frontiersin.org/articles/10.3389/fbioe.2018.00198/full